### PR TITLE
[CDF-21423] Added warning if action and shared are in a transformation yaml

### PR DESCRIPTION
**Pull request by doctrino**
# Description

Changed to error
```
ERROR: Failed to load 8.pump_asset_hierarchy-load-collections_pump.yaml with transformations. Error: Parameters invalid. These are specific for the `transformation-cli` and not supported by cognite-toolkit
In 'tr_pump_asset_hierarchy-load-collections_pump' the following parameters are invalid: 'action' should be 'conflictMode'.
```

## Checklist

- [ ] Tests added/updated.
- [ ] Run Demo Job Locally.
- [ ] Documentation updated.
- [ ] Changelogs updated in [CHANGELOG.cdf-tk.md](https://github.com/cognitedata/cdf-project-templates/blob/main/CHANGELOG.cdf-tk.md).
- [ ] Template changelogs updated in [CHANGELOG.templates.md](https://github.com/cognitedata/cdf-project-templates/blob/main/CHANGELOG.templates.md).
- [ ] Version bumped.
  [_version.py](https://github.com/cognitedata/cdf-project-templates/blob/main/cognite/cognite_toolkit/_version.py) and
  [pyproject.toml](https://github.com/cognitedata/cdf-project-templates/blob/main/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite_toolkit/_cdf_tk/load/_resource_loaders.py [hunk 1]
```diff
@@ -1023,6 +1023,19 @@ def load_resource(
         transformations = TransformationWriteList([])
 
         for resource in resources:
+            external_id = resource.get("externalId", "Missing")
```
Comment (haakonvt) [inline cognite_toolkit/_cdf_tk/load/_resource_loaders.py:1026]: Perhaps make it even more clear, I leave it up to you:
```suggestion
            external_id = resource.get("externalId", "<Missing>")
```

**Review by ronpal — APPROVED** [review]

**Review by haakonvt — APPROVED** [review]
I am saddened by the CLI not following the API spec... anyway this seems like very valuable feedback to the user. Should we raise? Given that we understand the intention of the user?

File: cognite_toolkit/_cdf_tk/load/_resource_loaders.py [hunk 2]
```diff
@@ -1023,6 +1024,19 @@ def load_resource(
         transformations = TransformationWriteList([])
 
         for resource in resources:
+            parameters: dict[str, str] = {}
```
Comment (haakonvt) [inline cognite_toolkit/_cdf_tk/load/_resource_loaders.py:1027]: Perhaps add invalid to the name:
```py
invalid_parameters: dict[str, str] = {}
invalid_params: dict[str, str] = {}
```

File: cognite_toolkit/_cdf_tk/load/_resource_loaders.py [hunk 3]
```diff
@@ -1023,6 +1024,19 @@ def load_resource(
...
+            if "action" in resource and "conflictMode" not in resource:
+                parameters["action"] = "conflictMode"
+            if "shared" in resource and "isPublic" not in resource:
+                parameters["shared"] = "isPublic"
+            if parameters:
+                raise ToolkitInvalidParameterError(
```
Comment (haakonvt) [inline cognite_toolkit/_cdf_tk/load/_resource_loaders.py:1033]: For consideration, to distinguish from an invalid value:
```py
raise ToolkitInvalidParameterNameError(
```

**Pull request merged**